packages/contracts/contracts/optimistic-ethereum/utils/libraries/TransactionParser.sol

```diff
@@ -58,4 +59,63 @@ library TransactionParser {
 
         return RLPWriter.encodeList(raw);
     }
+
+    function encodeEOATransaction(
+        DataTypes.EOATransaction memory _transaction,
+        bool _isEthSignedMessage
+    )
+        internal
+        pure
+        returns (
+            bytes memory
+        )
+    {
+        if (_isEthSignedMessage) {
+            return abi.encode(
+                _transaction.nonce,
+                _transaction.gasLimit,
+                _transaction.gasPrice,
+                _transaction.to,
+                _transaction.data
+            );
+        } else {
+            bytes[] memory raw = new bytes[](9);
+
+            raw[0] = RLPWriter.encodeUint(_transaction.nonce);
+            raw[1] = RLPWriter.encodeUint(_transaction.gasPrice);
+            raw[2] = RLPWriter.encodeUint(_transaction.gasLimit);
+            raw[3] = RLPWriter.encodeAddress(_transaction.to);
+            raw[4] = RLPWriter.encodeUint(0);
+            raw[5] = RLPWriter.encodeBytes(_transaction.data);
+            raw[6] = RLPWriter.encodeUint(108);
```

Comment:
Instead of hardcoding 108, shouldn't this use something like:

```solidity
uint256 id;
assembly {
    id := chainid()
}

raw[6] = RLPWriter.encodeUint(id);
```